browser/ui/views/infobars/brave_confirm_infobar.cc

```diff
@@ -0,0 +1,85 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/views/infobars/brave_confirm_infobar.h"
+
+#include <memory>
+#include <utility>
+
+#include "base/bind.h"
+#include "ui/base/metadata/metadata_impl_macros.h"
+#include "ui/views/controls/button/checkbox.h"
+
+namespace {
+
+constexpr int kCheckboxSpacing = 20;
+
+}  // namespace
+
+std::unique_ptr<infobars::InfoBar> CreateBraveConfirmInfoBar(
+    std::unique_ptr<BraveConfirmInfoBarDelegate> delegate) {
+  return std::make_unique<BraveConfirmInfoBar>(std::move(delegate));
+}
+
+BraveConfirmInfoBar::BraveConfirmInfoBar(
+    std::unique_ptr<BraveConfirmInfoBarDelegate> delegate)
+    : ConfirmInfoBar(std::move(delegate)) {
+  auto* delegate_ptr = GetBraveDelegate();
+  if (delegate_ptr->HasCheckbox()) {
+    checkbox_ = AddChildView(std::make_unique<views::Checkbox>(
+        delegate_ptr->GetCheckboxText(),
+        base::BindRepeating(&BraveConfirmInfoBar::CheckboxPressed,
+                            base::Unretained(this))));
+  }
+}
+
+BraveConfirmInfoBar::~BraveConfirmInfoBar() = default;
+
+void BraveConfirmInfoBar::Layout() {
+  ConfirmInfoBar::Layout();
+
+  // Early return when checkbox is not used.
+  // This class is only valid when this infobar has checkbox now.
+  // NOTE: Revisit when we want to use other buttons together with checkbox.
+  if (!checkbox_)
+    return;
+
+  checkbox_->SizeToPreferredSize();
+
+  DCHECK(!ok_button_ && !cancel_button_);
```

Comment:
nit: could we have a more info for this check? like,
<< "We don't consider case where ok/cancel button and check box exist together". 

And I think it'd be good to have this check in the constructor when creating check_box_.